provisioner/ansible/provisioner.go

```diff
@@ -382,6 +391,21 @@ func validateFileConfig(name string, config string, req bool) error {
 	return nil
 }
 
+func validateDirectoryConfig(name string, config string, req bool) error {
```

Comment:
What's the approach you want me to take here?  Should I just get rid of the `validateDirectoryConfig` function, and inline all of that logic in the `Prepare` function?